mongo/errors.go

```diff
@@ -116,6 +116,17 @@ func IsTimeout(err error) bool {
 	return false
 }
 
+// IsDeadlineExceededError returns true if err is from an expired or canceled context
```

Comment:
Should this be a private error helper instead? I am thinking:

1. It is not MongoDB specific error handling logic
2. Go 1.13+ users can use `errors.Is`

If so, I suggest making it a private helper (perhaps in session.go) and renaming it. Perhaps it could be: `isNonRetryableTransactionContextError`?